collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -372,7 +372,12 @@ enum AnimationType {
      */
     @Override
     public void onCreate(Bundle savedInstanceState) {
-        Timber.w("onCreate %s", Md5.getMd5Hash(getIntent().getData().toString()));
+        if (getIntent().getData() != null) {
```

Comment:
I'm not sure what you mean. Moving the logging there?

Reply:
We can move that check to `FormUriActivity` as `FormEntryActivity` is never directly started right? It's always through `FormUriActivity` so we can check for null there and not start `FormEntryActivity` at all.

Reply:
Ah I see. It looks like `InstanceChooserList` still starts `FormEntryActivity` directly. It looks like `FormEntryActivity` does handle a null data URI (https://github.com/getodk/collect/blob/4ced1468b013d976b175bba0394631cd0256113e/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L694), so I was thinking we just needed to prevent this particular crash.

Reply:
> It looks like InstanceChooserList still starts FormEntryActivity directly

so it would be good to fix this.

> It looks like FormEntryActivity does [handle a null data URI](https://github.com/getodk/collect/blob/4ced1468b013d976b175bba0394631cd0256113e/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L694), so I was thinking we just needed to prevent this particular crash.

So we could even remove that check from `FormEntryActivity` then.
 
To me `FormUriActivity` seems like a place where we should check things that are crucial, requirements that if are not met it doesn't make sense to start the form at all. For example, we check if projects exist in that class.

Reply:
Just to be clear, I wasn't trying to clean up or change the behaviour when the data URI is null. I was just trying to prevent the crash that the logging line was causing. I do think you're correct that we could be "sanitizing" the intent passed to `FormEntryActivity` with `FormUriActivity`, but that seems like a larger (and riskier) change that we don't need to make to prevent this crash.

Reply:
But this simple check will not suffice. If the data is null the same NPE will take place in `onStart` and then in `onResume` - both methods are called before the place you pointed out where we check if the data is null and display an error dialog.

Reply:
I understand that we don't have time in this cycle to work on a better solution so I"ve added missing null checks and merged the pr. I've also filed a separate issue https://github.com/getodk/collect/issues/5484 for what I said above.